lib/figma.js

```diff
@@ -45,3 +45,100 @@ exports.findAllTokens = (data, tokenType) =>
+ * to call the Figma client and receive the image urls.
+ *
+ * @param {Array<Object>} exportableData
+ * @return {Array<Object>}
+ */
+const parseImageRequestData = (exportableData) =>
+  exportableData.reduce((acc, component) => {
+      component.exportSettings.forEach((setting) => {
+        const { suffix } = setting;
+        const format = setting.format.toLowerCase();
+        let scale = 1;
+        if (setting.constraint && setting.constraint.type && setting.constraint.type === 'SCALE') {
+          scale = setting.constraint.value;
+        }
+
+        const existingElement = acc.find(c => c.format === format && c.scale === scale);
+        if (existingElement) {
+          existingElement.ids.push(component.id);
+        } else {
+          acc.push({
+            ids: [component.id],
+            format,
+            scale,
+            suffix,
+          })
+        }
+      });
+      return acc;
+    }, []);
+
+/**
+ * Calls the images endpoint of the figma api (using figma-js client).
+ * This call returns an images map where each key is an object id, and each
+ * value is a url of the image in the requested format.
+ * See https://www.figma.com/developers/api#get-images-endpoint for more info
+ *
+ * @param {Object} client
+ * @param {string} fileId
+ * @param {Object} imageRequest
+ * @return {Object}
+ */
+const getImageUrls = async (client, fileId, imageRequest) => {
+  const imageResponse = await client.fileImages(fileId, imageRequest);
+  return {
+    format: imageRequest.format,
+    suffix: imageRequest.suffix,
+    urls: Object.entries(imageResponse.data.images).map(([id, url]) => ({ id, url })),
+  };
+};
+
+/**
+ * Retrieves the urls for all images-exportsetting combinations defined in the (flattened)
+ * figma api response. Returns a map of url, id and full filename (including suffix and extension)
+ *
+ * @param {Object} client
+ * @param {string} fileId
+ * @param {Array<Object>} data
+ * @return {Array<Object>}
+ */
+exports.retrieveImageUrls = async(client, file, data) => {
+  try {
+    // filter out only components containing exportSettings
+    const exportableData = data.filter(component => component.exportSettings);
+
+    // Parse all necessary data for the images request to figma api
+    const imageRequestData = parseImageRequestData(exportableData);
+
+    // Request the image urls for each image-exportsetting combination
+    const imageRequestResults = await Promise.all(imageRequestData
+      .map(imageRequest => getImageUrls(client, file, imageRequest))
+    );
+
+    // Now create an array of objects containing the full desired filename and url.
+    return imageRequestResults.reduce((acc, result) => {
+      result.urls.forEach(({ id, url }) => {
+        const matchingFileData = exportableData.find(d => d.id === id);
+        acc.push({
+          id,
+          fileUrl: url,
+          fileName: `${matchingFileData.name}${result.suffix}.${result.format}`,
```

Comment:
Ensure that filenames are always lowercase snake cased. This is required for assets exported with Hubble.